src/Jobs/PendingTransitionsDispatcher.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+
+namespace Asantibanez\LaravelEloquentStateMachines\Jobs;
+
+
+use Asantibanez\LaravelEloquentStateMachines\Models\PendingTransition;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+
+class PendingTransitionsDispatcher implements ShouldQueue
+{
+    use InteractsWithQueue, Queueable, Dispatchable, SerializesModels;
+
+    public function handle()
+    {
+        PendingTransition::with(['model'])
+            ->notApplied()
+            ->onScheduleOrOverdue()
+            ->get()
+            ->each(function (PendingTransition $pendingTransition) {
+                PendingTransitionExecutor::dispatch($pendingTransition);
+
+                $pendingTransition->applied_at = now();
+                $pendingTransition->save();
```

Comment:
if there is a queue delay or backlog, this could give false results as to when it was applied maybe? I guess it would know when it was dispatched and the state history would know when it was actually applied so maybe it doesn't matter.

Reply:
Yes. Had the exact same thought. 

I think `pending_transitions` table should be purged automatically when dispatching the executor job. This way, the only timestamp that matters is in `state_histories`. 

What do you think?